src/System.Buffers.Primitives/System/Buffers/Pooling/ArrayMemoryPool.cs

```diff
@@ -10,12 +9,19 @@ namespace System.Buffers.Internal
 {
     internal sealed class ArrayMemoryPool<T> : MemoryPool<T>
     {
+        const int DefaultSize = 4096;
+
         readonly static ArrayMemoryPool<T> s_shared = new ArrayMemoryPool<T>();
 
         public static ArrayMemoryPool<T> Shared => s_shared;
 
-        public override OwnedMemory<T> Rent(int minimumBufferSize)
+        public override int MaxBufferSize => 1024 * 1024 * 1024;
+
+        public override OwnedMemory<T> Rent(int minimumBufferSize = DefaultSize)
```

Comment:
I don't think we should use default value here. Especially while having `AnySize` option.

Reply:
Default and Any are different. Default is a const. Any is a sentinel. Any can return different sizes each time it's called. Default will return the same size.

Reply:
But it get's compiled into consumer code. So if implementation changes it would require a recompile to react.

Reply:
I don't get it: who needs to react to what? If the calling code does not pass anything to Rent, it apparently does not care what size it gets back. 

Having said that, I will change it to Any to be consistent across all pools.

Reply:
Imagine we have pipes `MemoryPool` with block size `4096` and `minimumBufferSize` being the same. Then we ship a new version with smaller block size. While user code looks like it doesn't care about the renting size it would start failing until rebuilt against a newer version of the library.